samza-core/src/main/java/org/apache/samza/coordinator/JobCoordinator.java

```diff
@@ -55,12 +57,14 @@
    * @throws InterruptedException if the current thread is interrupted while waiting for the JobCoordinator to start-up
    */
   boolean awaitStart(long timeoutMs) throws InterruptedException;
+
   /**
-   * Returns the logical ID assigned to the processor
-   * It is up to the user to ensure that different instances of StreamProcessor within a job have unique processor ID.
-   * @return integer representing the logical processor ID
+   * Returns the logical ID assigned to the processor that is local to the instance of StreamProcessor.
+   * It is the responsibility of the JobCoordinator to assign a unique identifier based on the underlying environment.
```

Comment:
>> It is the responsibility of the JobCoordinator to assign a unique identifier based on the underlying environment

- Move it to the docs of the class instead of documenting it here? It we have already called out, this is probably redundant

- Will also be helpful to state the requirements of the ID here.